### ISSUE TITLE
FNX3-16660 ⁃ Add action states to share sheet

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/fenix#2897]
As a user, I want to be able to initiate the following actions from the share sheet:

* "Verify your email address" - if a user has signed up for an account but hasn't verified their email address during that process before they can make use of sync.
* "Connect another device" - user has signed into their account but doesn't have other devices syncing to send a tab to.
* "Reconnect to Sync" - error occurred within user account and user needs to re-login.
* "Offline" - No network connection.

Mocks: https://mozilla.invisionapp.com/share/QSR2F8UD4PN#/screens/374762782

Acceptance criteria:
-If I have signed up for an account but haven't yet verified my email address, I will be see a "Verify your email address" icon when I open the share sheet. 
-If I have signed into my account but don't have other devices syncing, I will see a "Connect another device" icon when I open the share sheet. 
~~If there is an error within my account sync, I will be notified that I need to re~~login by seeing a "Reconnect to Sync" icon when I open the share sheet. 
-If I don't have network connection and try to share something, I will see the "Offline" icon when I open the share sheet.

**Comment by jonalmeida** [comment]
Related to https://github.com/mozilla-mobile/fenix/issues/2897 and depends on the same fixes.

**Comment by jonalmeida** [comment]
Adding a-c label since this may involve some a-c changes as well before we can consume the a-s bits.

**Comment by jonalmeida** [comment]
@vrudesai I've changed the feature label since this this isn't related to the QAB directly.

@vesta0 if we want to do this for Post-MVP, there is still a small amount of work to hide the send-tab section if we encounter an error. Shall I file another issue for that?

**Comment by liuche** [comment]
We're looking at this during Epic sizing, and @vesta0 could you add some Acceptance Criteria - it's not clear to me what needs to be added.

**Comment by mheubusch** [comment]
@jonalmeida can you describe the errors to me so I can write strings? Do you just need the label under the two states above?

**Comment by jonalmeida** [comment]
@mheubusch I've updated the title to say 'action states' since they aren't all errors.

For these action states there aren't any messages we want to display other than the error title. You can see the user flows here: https://mozilla.invisionapp.com/share/QSR2F8UD4PN#/screens/364885746

The ones for this ticket are:
- "Verify your email address" - if a user has signed up for an account but hasn't verified their email address during that process before they can make use of sync.
- "Connect another device" - we have an account, my no other devices to send to.
- "Reconnect to Sync" - and error occurred within your account and we need you to re-enter your login.
- "Offline" - No network connection.

**Comment by mheubusch** [comment]
@jonalmeida still not sure what the request is - do you want me to approve the copy @ryanfeeley has in the designs (yellow bars with errors or snackbar messages)?

**Comment by jonalmeida** [comment]
> @jonalmeida still not sure what the request is - do you want me to approve the copy @ryanfeeley has in the designs (yellow bars with errors or snackbar messages)?

Sorry Michelle, these designs have changed again since the original screenshot above.

**Comment by jonalmeida** [comment]
There isn't any a-c work required here any more.

**Comment by vesta0** [comment]
@vesta0 will re-confirm states and acceptance criteria with @ryanfeeley

**Comment by vesta0** [comment]
I cross-referenced the acceptance criteria with the most recently updated mocks referenced [here](https://github.com/mozilla-mobile/fenix/issues/3077#issuecomment-515018983) and it looks aligned to me.

**Comment by colintheshots** [comment]
UX: I assume we should build the feature as designed, except leave out the "Verify your email address" portion as we can't support it today?

**Comment by ekager** [comment]
I picked up this ticket and since we don't have "Verify your email address" as an exposed state and we already had "Connect another device", I'm just adding:
- "Reconnect to Sync" - error occurred within user account and user needs to re-login.
- "Offline" - No network connection.

**Comment by ekager** [comment]
@ryanfeeley I'm not sure if the invision mocks linked in the original comment are the most up to date, but I tagged you in a question there about the "Offline" state. :)

**Comment by sblatz** [comment]
@sv-ohorvath Does this have bugs? Confused why the `qa-needed` label was removed 😅

**Comment by sv-ohorvath** [comment]
@sblatz I've found https://github.com/mozilla-mobile/fenix/issues/5258  and set this one as blocked by it. So this doesn't need qa anymore, until the blocking bug is fixed.

**Comment by ekager** [comment]
I have merged a patch for #5258 so when that is verified, I think we can close this

**Comment by sv-ohorvath** [comment]
Blocking issue was fixed. Closing this issue.
